test/integration/test_dataset.py

```diff
@@ -34,13 +34,32 @@ def test_populate(conservator):
     assert dataset_from_id.name == dataset.name
 
 
-@pytest.mark.skip()
 def test_delete(conservator):
     dataset = conservator.datasets.create("My dataset")
-    assert conservator.datasets.count() == 1
+    # Give the server time to create and commit the dataset.
+    found = False
+    for _ in range(10):
+        time.sleep(1)
+        for dset in conservator.datasets.all():
+            if dset.id == dataset.id:
+                found = True
+                break
+        if found:
+            break
+    assert found
 
     dataset.delete()
-    assert conservator.datasets.count() == 0
+    # Give the server time to commit and delete the dataset.
+    found = False
+    for _ in range(10):
+        time.sleep(1)
+        for dset in conservator.datasets.all():
```

Comment:
This could, likewise, just query for the one dataset by ID, which should return a result until the delete process is finished. 10 seconds may or may not be enough for that to complete either.

Reply:
The integration tests are run on a local Kind instance, so there shouldn't be any other activity slowing down the server and few, if any other datasets in the search results.  I'm surprised to hear that deleting an empty dataset under those conditions could take longer than 10 seconds.  When I ran the integration tests locally, I saw errors until I added the delays, which tells me there's a possible race condition.  Without visibility into what's happening on the server side, adding delays is the simplest method to handle those.

Reply:
Added a new commit that checks that the dataset ID becomes invalid after deleting the dataset.